packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -26,10 +26,30 @@ import {
 
 // Truthiness utilities
 // #region
+const valueIsPseudoBigInt = (
+  value: number | string | ts.PseudoBigInt,
+): value is ts.PseudoBigInt => {
+  return typeof value === 'object' && 'base10Value' in value;
+};
+
+const getValue = (type: ts.LiteralType): bigint | number | string => {
+  if (valueIsPseudoBigInt(type.value)) {
+    return BigInt(type.value.base10Value);
```

Comment:
I don't think it will make a difference in correctness for this PR but probably best to be precise about negative numbers. 
```suggestion
    return BigInt((type.value.negative ? '-' : '') + type.value.base10Value);
```
See also https://github.com/typescript-eslint/typescript-eslint/pull/10194/files#diff-3cc43b506517aba406ae3a31d7998b116c4b09db76b632c6cb3c2e54fc71d66cR72-R93 (not merged yet) that we'll probably want to align on